arangod/Aql/JoinNode.cpp

```diff
@@ -147,6 +147,8 @@ void JoinNode::doToVelocyPack(VPackBuilder& builder, unsigned flags) const {
     // condition
     builder.add(VPackValue("condition"));
     it.condition->toVelocyPack(builder, flags);
+    // projections
+    it.projections.toVelocyPack(builder);
```

Comment:
This is fine for serialization, but AFAICS the generated "projections" velocypack attribute is never read back.
The JoinNode constructor that is supposed to create a JoinNode from a velocypack Slice should read the "projections" attribute from each IndexInfo entry somehow (e.g. here: https://github.com/arangodb/arangodb/pull/19895/files#diff-b284b202b1037fe300a1226b5a94c5b844d63bfbbfcaa3d42af091075346da6bR123-R127).
Otherwise the packaged projections info will be lost.

Reply:
Will be done in a follow up PR, when I prepare the JoinNode for execution in the cluster.